File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptDecimalValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptDecimalInValue.OptDecimalValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptDecimalValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptDecimalValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                java.math.BigDecimal.class,
                OptDecimalValue.class.getDeclaredField("optDecimal").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                java.math.BigDecimal.class,
                OptDecimalValue.class.getMethod("getOptDecimal").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptDecimalValue.class,
                OptDecimalValue.class.getMethod("setOptDecimal", java.math.BigDecimal.class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptDecimalValue().getOptDecimal());
    }
}

File: legacy/workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptDecimalValue.scala
package com.dslplatform.ocd.values

import OptDecimalInValue.OptDecimalValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptDecimalValue extends SpecExtensions {
  def `Test property field type: Option[BigDecimal]`() =
    checkType(
      typeOf[Option[BigDecimal]]
    , typeOf[OptDecimalValue].member("optDecimal": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptDecimalValue().optDecimal === None)
  }
}
